addons/hr_holidays/views/hr_leave_views.xml

```diff
@@ -358,6 +364,13 @@
             <xpath expr="//div[hasclass('o_hr_leave_column')]" position="attributes">
                 <attribute name="class" remove="col_left col-md-6" separator=" "/>
             </xpath>
+            <xpath expr="//sheet/widget[@name='web_ribbon']" position="after">
+                <widget name="web_ribbon" title="Draft" bg_color="bg-warning" attrs="{'invisible': [('state', '!=', 'draft')]}"/>
+                <widget name="web_ribbon" title="To Approve" bg_color="bg-info" attrs="{'invisible': [('state', '!=', 'confirm')]}"/>
+                <widget name="web_ribbon" title="Refused" bg_color="bg-danger" attrs="{'invisible': [('state', '!=', 'refuse')]}"/>
+                <widget name="web_ribbon" title="Second Approval" bg_color="bg-info" attrs="{'invisible': [('state', '!=', 'validate1')]}"/>
+                <widget name="web_ribbon" title="Approved" bg_color="bg-success" attrs="{'invisible': [('state', '!=', 'validate')]}"/>
```

Comment:
Maybe it would also be interesting to hide the ribbons while the user is creating the leave, I think it can be done by adding `('id', '=', False)` in the domain